dev/src/dev/rewrite_defendpoint_1.clj

```diff
@@ -0,0 +1,422 @@
+(ns dev.rewrite-defendpoint-1
+  (:require
+   [clojure.string :as str]
+   [flatland.ordered.map :as ordered-map]
+   [metabase.util :as u]
+   [metabase.util.files :as u.files]
+   [metabase.util.malli :as mu]
+   [metabase.util.malli.registry :as mr]
+   [metabase.util.malli.schema :as ms]
+   [rewrite-clj.node :as n]
+   [rewrite-clj.parser :as r.parser]
+   [rewrite-clj.zip :as z]))
+
+(set! *warn-on-reflection* true)
+
+(mr/def ::zloc vector?)
+
+(mr/def ::node
+  [:fn
+   {:error/message "Valid rewrite-clj node"}
+   (fn node? [x]
+     (and
+      (n/node? x)
+      (or (not (n/inner? x))
+          (every? node? (n/children x)))))])
+
+(mr/def ::schema-map
+  "Map of raw symbol => schema node."
+  [:maybe
+   [:map-of symbol? ::node]])
+
+(mr/def ::parsed
+  [:map
+   [:route      ::node]
+   [:method     ::node]
+   [:fn-args    ::node]
+   [:schema-map ::schema-map]])
+
+(mu/defn- parse-defendpoint-1-args :- ::parsed
+  [defendpoint :- ::zloc]
+  (let [symb       (z/down defendpoint)
+        method     (z/right symb)
+        route      (z/right method)
+        args       (z/right route)
+        ;; skip over docstring
+        fn-args    (if (string? (z/sexpr args))
+                     (z/right args)
+                     args)
+        args       (z/right fn-args)
+        schema-map (when (and (= (z/tag args) :map)
+                              ;; there has to be more stuff after this
+                              (z/right args))
+                     (into (ordered-map/ordered-map)
```

Comment:
I'm using `ordered-map` to preserve the order of the arguments from the original schema maps in the new `:map` schemas